src/api/c++/z3++.h

```diff
@@ -448,14 +506,23 @@ namespace z3 {
     public:
         param_descrs(context& c, Z3_param_descrs d): object(c), m_descrs(d) { Z3_param_descrs_inc_ref(c, d); }
         param_descrs(param_descrs const& o): object(o.ctx()), m_descrs(o.m_descrs) { Z3_param_descrs_inc_ref(ctx(), m_descrs); }
+        param_descrs(param_descrs && o) noexcept : object{std::forward<object>(o)}, m_descrs{o.m_descrs} {
+            o.m_descrs = nullptr;
+        }
         param_descrs& operator=(param_descrs const& o) {
             Z3_param_descrs_inc_ref(o.ctx(), o.m_descrs);
             Z3_param_descrs_dec_ref(ctx(), m_descrs);
             m_descrs = o.m_descrs;
-            m_ctx = o.m_ctx;
+            object::operator=(o);
             return *this;
         }
-        ~param_descrs() { Z3_param_descrs_dec_ref(ctx(), m_descrs); }
+        param_descrs& operator=(param_descrs && o) noexcept {
+            object::operator=(std::forward<object>(o));
```

Comment:
std::move

Reply:
`std::forward`'s advantage here is that it also explicitly up-casts, but functionally that should be the same; this is just not-implict. If it's a sticking point though I can change it, I just prefer explicit casting to implicit casting as a convention.

Reply:
I don't see any advantage. It takes more bytes and gives no benefit.

Reply:
I'm resolving this and the other issues that are duplicates of the one above to consolidate the conversation.